### ISSUE TITLE
SUBSCRIBE: WHILE ZIPPING

### ISSUE THREAD

**Issue by baxri** [mockingbot/react-native-zip-archive#104]
## MAY BE A BUG

I am trying zip folder and show progress bar to user, but subscribe event runs only once and `progress` value is 0 and `filePath` is `undefined` . Problem is on Android OS 7.0, my code looks like this: 

```javascript

import { zip, unzip, unzipAssets, subscribe } from 'react-native-zip-archive'

subscribe(({ progress, filePath }) => {
    console.log(progress); // 0
    console.log(filePath); // undefined
})

let targetPath = PATH_BACKUP + '/' + zipName;
let sourcePath = PATH;
    
zip(sourcePath, targetPath).then(path => {

}).cath(error => {

});

```

**Comment by baxri** [comment]
Is there any news on this issue?

**Comment by baxri** [comment]
@plrthink I noticed one more thing, the callback is called only twice while zipping when progress is 0 and then when progress is 1.

**Comment by baxri** [comment]
@plrthink I updated package to 3.0.0 and now subscribe callback while zupping is works it always shows 1

```javascript
{progress: 0, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}
{progress: 1, filePath: "/storage/emulated/0/Download/BACKUPS/34.zip"}

```

**Comment by plrthink** [comment]
Sorry for the late response, but not sure what's the problem. Will dig into it.

**Comment by baxri** [comment]
@plrthink no problem :) the last issue is that subscribe callback always returns progress 1 while zipping.

**Comment by baxri** [comment]
Hey @plrthink. The problem is while **zipping**, unzip is working greate.

environment information:

```
Environment:
  OS: macOS High Sierra 10.13.6
  Node: 8.11.2
  Yarn: Not Found
  npm: 5.6.0
  Watchman: Not Found
  Xcode: Xcode 10.0 Build version 10A255
  Android Studio: 3.1 AI-173.4720617

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.3 => 0.55.3
  react-native-zip-archive: ^3.0.0

```

zipping code looks like this:

```javascript
let zipped = await zip(sourcePath, targetPath);
```

and this is subscribe

```javascript
       componentDidMount() {
        this.zipProgress = subscribe((objectProgress) => {
            console.log(objectProgress);
            this.setState({ zipProgress: objectProgress.progress })
        })
   
```

**Comment by plrthink** [comment]
@baxri , oh, now I can see the problem.

Can you test on this branch? https://github.com/mockingbot/react-native-zip-archive/tree/fix-zip-progress.  If it works I would release a patch ASAP.

**Comment by baxri** [comment]
@plrthink 

I installed this branch with this command

```npm install --save mockingbot/react-native-zip-archive#fix-zip-progress```

But now I am getting this excepption

```
Start Zipping

// DEST
/storage/emulated/0/Download/BACKUPS/eee.zip

//SOURCE
/storage/emulated/0/Pictures/HACCPIMAGES-8

D:\react\haccp\src\screens\admin\backup\index.js:170 Error: Uncaught exception in ZipTask: java.lang.NoClassDefFoundError: Failed resolution of: Ljava/nio/file/Paths;
    at createErrorFromErrorData (D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\NativeModules.js:121)
    at D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\NativeModules.js:78
    at MessageQueue.__invokeCallback (D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:398)
    at D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:137
    at MessageQueue.__guardSafe (D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:314)
    at MessageQueue.invokeCallbackAndReturnFlushedQueue (D:\react\haccp\node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:136)
    at debuggerWorker.js:70
```

**Comment by plrthink** [comment]
@baxri have updated to address this compatible issue, please try again.

**Comment by baxri** [comment]
@plrthink That's it works great!!! :)

**Comment by baxri** [comment]
@plrthink Thank you for your support 👍

**Comment by plrthink** [comment]
@baxri just released 3.0.1 to solve this.
